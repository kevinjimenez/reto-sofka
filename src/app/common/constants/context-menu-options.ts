import { ContextMenuOption } from '../../shared/components/context-menu/interfaces/context-menu-option.interface';

export const CONTEXT_MENU_OPTIONS: ContextMenuOption[] = [
	{
		label: 'Actualizar',
		value: 1
	},
	{
		label: 'Eliminar',
		value: 2
	}
];
