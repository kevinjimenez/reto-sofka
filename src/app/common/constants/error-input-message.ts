export const ERROR_INPUT_MESSAGE: Record<string, string> = {
	required: 'El campo {{label}} es requerido',
	email: 'El campo {{label}} no es un correo valido',
	password:
		'El campo {{label}} debe tener al menos una minuscula, mayuscula, número, caracter especial y longitud sea mayor o igual a 4',
	minlength: 'El campo {{label}} debe tener minimo {{minlength}} caracteres',
	maxlength: 'El campo {{label}} debe tener maximo {{maxlength}} caracteres',
	currentDate: 'La fecha debe ser mayor o igual a la fecha actual',
	oneYearAfterRelease: 'La fecha debe ser mayor o igual a la fecha actual + 1 año',
	availableId: 'El ID ya existe'
};
