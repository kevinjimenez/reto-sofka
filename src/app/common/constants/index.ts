export { CONTEXT_MENU_OPTIONS } from './context-menu-options';
export { PRODUCT_TABLE_HEADERS } from './table-headers';
export { ERROR_INPUT_MESSAGE } from './error-input-message';
