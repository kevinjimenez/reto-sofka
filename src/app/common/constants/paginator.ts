export const PAGINATOR = {
	default: '5',
	range: ['5', '10', '15']
};
