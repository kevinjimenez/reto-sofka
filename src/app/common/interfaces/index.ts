export { Product } from './product.interface';
export { DefaultResponse } from './default-response.interface';
export { TableHeaderInterface } from './table-header.interface';
export { ErrorInputInterface } from './error-input.interface';
