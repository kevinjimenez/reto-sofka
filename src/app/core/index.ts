export { spinnerInterceptor } from './interceptors/spinner.interceptor';
export { productsResolver } from './resolvers/products.resolver';
export { ProductsService } from './services/products.service';
export { SpinnerService } from './services/spinner.service';
