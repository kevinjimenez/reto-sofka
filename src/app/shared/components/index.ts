export { ContextMenuOption } from './context-menu/interfaces/context-menu-option.interface';
export { ContextMenuOptionSelected } from './context-menu/interfaces/context-menu-option-selected.interface';
export { ContextMenuComponent } from './context-menu/context-menu.component';
export { IconButtonComponent } from './icon-button/icon-button.component';
export { AlertCircleComponent } from './icons/alert-circle/alert-circle.component';
export { OverlayComponent } from './overlay/overlay.component';
export { SpinnerComponent } from './spinner/spinner.component';
export { ButtonComponent } from './button/button.component';
export { InputComponent } from './input/input.component';
export { LogoComponent } from './logo/logo.component';
export { ToastComponent } from './toast/toast.component';
export { TooltipComponent } from './tooltip/tooltip.component';
