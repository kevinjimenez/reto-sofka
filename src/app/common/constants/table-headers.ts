import { TableHeaderInterface } from '../interfaces';

export const PRODUCT_TABLE_HEADERS: TableHeaderInterface[] = [
	{ title: 'Logo' },
	{ title: 'Nombre del producto' },
	{
		title: 'Descripción',
		description: 'Descripción',
		tooltip: true
	},
	{
		title: 'Fecha de liberación',
		description: 'Fecha de liberación',
		tooltip: true
	},
	{
		title: 'Fecha de reestructuración',
		description: 'Fecha de reestructuración',
		tooltip: true
	},
	{ title: '' }
];
