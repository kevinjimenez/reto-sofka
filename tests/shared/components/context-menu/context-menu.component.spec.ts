import { ComponentRef } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { ContextMenuComponent } from '../../../../src/app/shared/components';
import { CONTEXT_MENU_OPTIONS } from '../../../../src/app/common/constants';
import { Product } from '../../../../src/app/common/interfaces';

describe('ContextMenuComponent', () => {
	let component: ContextMenuComponent;
	let componentRef: ComponentRef<ContextMenuComponent>;
	let fixture: ComponentFixture<ContextMenuComponent>;
	let compiled: HTMLElement;

	beforeEach(async () => {
		await TestBed.configureTestingModule({
			imports: [ContextMenuComponent]
		}).compileComponents();
	});

	beforeEach(() => {
		fixture = TestBed.createComponent(ContextMenuComponent);
		component = fixture.componentInstance;
		componentRef = fixture.componentRef;
		fixture.detectChanges();
		compiled = fixture.nativeElement;
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('should match snapshot', () => {
		expect(compiled).toMatchSnapshot();
	});

	it('should not render the context menu when isVisible is false', () => {
		component.isVisible.set(false);
		fixture.detectChanges();

		expect(compiled.querySelector('.context-menu')).toBeNull();
	});

	it('should hide render the context menu', () => {
		componentRef.setInput('options', CONTEXT_MENU_OPTIONS);
		component.isVisible.set(true);
		fixture.detectChanges();

		expect(compiled.querySelector('.context-menu')).toBeTruthy();

		component.hide();
		fixture.detectChanges();

		expect(compiled.querySelector('.context-menu')).toBeNull();
	});

	it('should render the context menu', () => {
		componentRef.setInput('options', CONTEXT_MENU_OPTIONS);
		component.isVisible.set(true);
		fixture.detectChanges();

		expect(compiled.querySelector('.context-menu')).toBeTruthy();
	});

	it('should emit optionSelected when an option is selected', () => {
		jest.spyOn(component.optionSelected, 'emit');
		component.itemSelected.set({ id: '1', name: 'Test Item' } as Product);
		component.onOptionSelected(0);
		fixture.detectChanges();

		expect(component.optionSelected.emit).toHaveBeenCalledWith({
			option: 0,
			item: component.itemSelected()
		});
	});

	it('should hide the context menu when document is clicked', () => {
		componentRef.setInput('options', CONTEXT_MENU_OPTIONS);
		component.isVisible.set(true); // Aseg√∫rate de que es visible
		fixture.detectChanges();

		component.onDocumentClick(new MouseEvent('click'));
		expect(component.isVisible()).toBeFalsy();
	});
});
