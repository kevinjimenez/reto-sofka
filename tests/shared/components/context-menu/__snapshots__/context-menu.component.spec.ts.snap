// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContextMenuComponent should match snapshot 1`] = `
<div
  id="root1"
>
  
</div>
`;
